types/header.go

```diff
@@ -65,7 +55,7 @@ type VTHeaderExtension struct {
 // )
 
 type HeaderExtensionEnum struct {
-	V2 V2HeaderExtension `json:"V2"`
+	V3 V3HeaderExtension `json:"V3"`
```

Comment:
Here i'm not sure if it actually works, because of enum discriminant, V3HeaderExtension should be at index 2 like we did for avail-js or avail-indexer. If it works it's ok, if not you can put 2 dummy values (or twice the V3HeaderExtension) then V3HeaderExtension

Reply:
yeh so, we are enforcing the encoder to take index value 2 as go has no enum :(